### DISCUSSION TITLE
Checking query data for undefined with Array.some does not satisfy Typescript

### DISCUSSION THREAD

**Discussion by m-fraczek** [TanStack/query#7238] [Q&A] [answered]
Posting it here because I'm not sure if it's an issue or it works as it should.
I'm using multiple useQuery hooks in my component. I check if any of the data is undefined by using Array.some and yet Typescript still shows an error 'Data is possibly undefined'. Here's an example: 

 ```
  const assets = getAssets();
  const collection = getCollection();
  const wishlist = getWishlist();

  if ([assets, collection, wishlist].some(q => !q.data)) return <Loading />;
```

Replacing condition with 
`!assets.data || !collection.data || !wishlist.data` 
Fixes the issue.

**Comment by TkDodo** [comment, marked as answer]
TypeScript does no type narrowing with array methods
